### ISSUE TITLE
add new commands that mimic unix cp/mv

### ISSUE THREAD

**Issue by ivandeex** [rclone/rclone#4777]
`rclone copy` and `rclone move` diverge in syntax and semantics from their unix counterparts. There are good reasons (efficiency and primary rclone function as a cloud tool) for why rclone move and rclone copy behave the way they do at the moment. Changing them would break many rclone based workflows in the wild.

Let's add new `rclone mv` and `rclone cp` which behave exactly like their unix counterparts. One of the things that would be required for these is multiple source parameters.
Originally submitted by @ncw at https://github.com/rclone/rclone/issues/4308#issuecomment-642259366

The all new `rclone cp` would support a flag like `--recurse` (or even `-R`) with semantics close to `cp -R` (TBD).
Originally submitted by @mrmcmuffinz at https://github.com/rclone/rclone/issues/1228#issuecomment-426869102

**Comment by GG-Sheridan** [comment]
Should this issue be labeled as Good First Issue?

We are doing research about how to help newcomers onboarding in an open source project and how to accelerate the issue resolving process.
We have developed a tool to automatically label the Good First Issue (GFI), i.e., the issue which is good for newcomers. 
We first extract 72 features from different dimensions of issue reports, such as readability, semantic meaning, associated code snippets, etc, and experimentally investigate to what extent these features correlate with GFI. Results turn out that a large portion of these features do have a medium to large correlation with GFI. 
We then build a machine learning model based on these features to predict GFI. The experiments on a dataset of 12k issues from Github show that 0.90 AUC can be achieved. 

Our tool shows that this issue has a very large possibility to be a GFI. 
Could you please tell us whether it should be labeled as a GFI?
And if you have any comments or suggestions with this tool, could you please tell us? 
Thank you so much.

**Comment by ncw** [comment]
@GG-Sheridan 
> Should this issue be labeled as Good First Issue?

No. It is quite tricky to replicate the features of `cp`/`mv` and to do so properly would require a lot of knowledge of rclone internals too.

**Comment by ivandeex** [comment]
Rephrasing requirements from #1175 in context of this feature request:

Currently `moveto` and `copyto` do not support moving files to an existing directory.
On the contrary, `rclone mv/cp` should re-create the semantics of `mv` and `cp`. Essentially these should move the file(s) into exsiting directory:
```
rclone mv file... existingdir
rclone mv file... existingdir/
```

(`moveto` would delete empty existing dir and rename `file` to that name, or fail if it's not empty)

**Comment by ivandeex** [comment]
From #2629 

As per [documentation](https://rclone.org/commands/rclone_copy) the following sequence of commands will not try to copy empty directory, just print an informational message
```
mkdir -p src/a
mkdir dst
rclone copyto src/a dst/a
```

On the contrary, the command `rclone cp --recursive src/a dst/a` should rather create empty directory on target side (the flag name is tentative).

**Comment by ivandeex** [comment]
Adding more requirements for the new commands from feature request #4308:

Currently rclone treats metacharacters (`*?`) in source file name as part of the name. The new commands will treat them as _globs_ similar to source filter arguments (`--include --exclude`).

There is a trick, though. If rclone is run interactively Unix(bash/*sh), user must properly escape metachars to prevent shell globbing (and let rclone do the **local** or remote listing with globbing/filtering and apply many other filtering options). If source is **local** then shell globbing would be _probably_ ok, but rclone would have to reparse command line (some verbs require more arguments than just source and destination).

**Comment by Exagone313** [comment]
I think rclone should have a command that has a behavior similar to rsync. The latter has some specifics about trailing slashes that are important to decide how to transfer files.

**Comment by davinkevin** [comment]
**question** is this still plan at some point? I use `rclone` as an `rsync` replacement, and it's a pain to deal with this limitation. 

Currently, I run something like this, kind hacky but it does the job: 

```
fd --exact-depth 1 folderS-I-want --threads=1 --exec rclone copyto --progress {} <dest>:/remote-folder/
```

However, `rclone` is damn good, and just the fact to have the config saved somewhere is for me the killing feature.

**Comment by ncw** [comment]
> Currently, I run something like this, kind hacky but it does the job:
> 
> ```
> fd --exact-depth 1 folderS-I-want --threads=1 --exec rclone copyto --progress {} <dest>:/remote-folder/
> ```

Can you @davinkevin explain exactly what that does? I can then see if I can replicate with rclone as running lots of individual rclone commands is inefficient and you are losing the opportunity for concurrency.

**Comment by davinkevin** [comment]
In a folder with 10 subfolders, I would like to `copy` 3 of them, all starting with the same prefix (`folderS-I-want` in the previous command). 

With the `fd --exact-depth 1 folderS-I-want` I select everything I want to transfer, and with the sub-sequent `--exec rclone copyto --progress {} <dest>:/remote-folder/` I launch the transfer of each… sequentially. 

So yes, it's less performant, but I think I have no choice at the moment, we can say to `rclone` to transfer what we point in the `cli` (which is the purpose of this issue).
